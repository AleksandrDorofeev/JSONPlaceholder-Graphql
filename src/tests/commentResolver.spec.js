const { createTestClient } = require("apollo-server-testing");
const {
  ApolloServer,
  makeExecutableSchema,
  gql,
} = require("apollo-server-express");

const typeDefs = require("../schema");
const resolvers = require("../resolver");

const GET_ALL_COMMENTS = gql`
  query GetAllComments {
    comments {
      postId
      id
      name
      email
      body
    }
  }
`;

const GET_ONE_COMMENT = gql`
  query GetOneComment($id: ID!) {
    comment(id: $id) {
      ... on CommentNotFound {
        error
        message
      }
      ... on Comment {
        postId
        id
        name
        email
        body
      }
    }
  }
`;

const constructServer = () => {
  const schema = makeExecutableSchema({
    typeDefs,
    resolvers,
    resolverValidationOptions: {
      requireResolversForResolveType: false,
    },
  });
  const server = new ApolloServer({ schema });
  return { server };
};

describe("[testing photo queries]", () => {
  it("GET_ALL_COMMENTS", async () => {
    const { server } = constructServer();
    const { query } = createTestClient(server);

    const res = await query({
      query: GET_ALL_COMMENTS,
    });

    expect(res).toMatchSnapshot();
  });

  it("GET_ONE_COMMENT with existing comment", async () => {
    const { server } = constructServer();

    const { query } = createTestClient(server);

    const res = await query({
      query: GET_ONE_COMMENT,
      variables: { id: 1 },
    });

    expect(res).toMatchSnapshot();
  });

  it("GET_ONE_COMMENT with not existing comment", async () => {
    const { server } = constructServer();

    const { query } = createTestClient(server);

    const res = await query({
      query: GET_ONE_COMMENT,
      variables: { id: 1000 },
    });

    expect(res).toMatchSnapshot();
  });
});
