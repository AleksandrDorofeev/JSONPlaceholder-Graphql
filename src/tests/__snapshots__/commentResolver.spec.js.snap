// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[testing photo queries] GET_ALL_COMMENTS 1`] = `
Object {
  "data": Object {
    "comments": Array [
      Object {
        "body": "laudantium enim quasi est quidem magnam voluptate ipsam eos
tempora quo necessitatibus
dolor quam autem quasi
reiciendis et nam sapiente accusantium",
        "email": "Eliseo@gardner.biz",
        "id": "1",
        "name": "id labore ex et quam laborum",
        "postId": "1",
      },
      Object {
        "body": "est natus enim nihil est dolore omnis voluptatem numquam
et omnis occaecati quod ullam at
voluptatem error expedita pariatur
nihil sint nostrum voluptatem reiciendis et",
        "email": "Jayne_Kuhic@sydney.com",
        "id": "2",
        "name": "quo vero reiciendis velit similique earum",
        "postId": "1",
      },
      Object {
        "body": "quia molestiae reprehenderit quasi aspernatur
aut expedita occaecati aliquam eveniet laudantium
omnis quibusdam delectus saepe quia accusamus maiores nam est
cum et ducimus et vero voluptates excepturi deleniti ratione",
        "email": "Nikita@garfield.biz",
        "id": "3",
        "name": "odio adipisci rerum aut animi",
        "postId": "1",
      },
      Object {
        "body": "non et atque
occaecati deserunt quas accusantium unde odit nobis qui voluptatem
quia voluptas consequuntur itaque dolor
et qui rerum deleniti ut occaecati",
        "email": "Lew@alysha.tv",
        "id": "4",
        "name": "alias odio sit",
        "postId": "1",
      },
      Object {
        "body": "harum non quasi et ratione
tempore iure ex voluptates in ratione
harum architecto fugit inventore cupiditate
voluptates magni quo et",
        "email": "Hayden@althea.biz",
        "id": "5",
        "name": "vero eaque aliquid doloribus et culpa",
        "postId": "1",
      },
      Object {
        "body": "doloribus at sed quis culpa deserunt consectetur qui praesentium
accusamus fugiat dicta
voluptatem rerum ut voluptate autem
voluptatem repellendus aspernatur dolorem in",
        "email": "Presley.Mueller@myrl.com",
        "id": "6",
        "name": "et fugit eligendi deleniti quidem qui sint nihil autem",
        "postId": "2",
      },
      Object {
        "body": "maiores sed dolores similique labore et inventore et
quasi temporibus esse sunt id et
eos voluptatem aliquam
aliquid ratione corporis molestiae mollitia quia et magnam dolor",
        "email": "Dallas@ole.me",
        "id": "7",
        "name": "repellat consequatur praesentium vel minus molestias voluptatum",
        "postId": "2",
      },
      Object {
        "body": "ut voluptatem corrupti velit
ad voluptatem maiores
et nisi velit vero accusamus maiores
voluptates quia aliquid ullam eaque",
        "email": "Mallory_Kunze@marie.org",
        "id": "8",
        "name": "et omnis dolorem",
        "postId": "2",
      },
      Object {
        "body": "sapiente assumenda molestiae atque
adipisci laborum distinctio aperiam et ab ut omnis
et occaecati aspernatur odit sit rem expedita
quas enim ipsam minus",
        "email": "Meghan_Littel@rene.us",
        "id": "9",
        "name": "provident id voluptas",
        "postId": "2",
      },
      Object {
        "body": "voluptate iusto quis nobis reprehenderit ipsum amet nulla
quia quas dolores velit et non
aut quia necessitatibus
nostrum quaerat nulla et accusamus nisi facilis",
        "email": "Carmen_Keeling@caroline.name",
        "id": "10",
        "name": "eaque et deleniti atque tenetur ut quo ut",
        "postId": "2",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`[testing photo queries] GET_ONE_COMMENT with existing comment 1`] = `
Object {
  "data": Object {
    "comment": Object {
      "body": "laudantium enim quasi est quidem magnam voluptate ipsam eos
tempora quo necessitatibus
dolor quam autem quasi
reiciendis et nam sapiente accusantium",
      "email": "Eliseo@gardner.biz",
      "id": "1",
      "name": "id labore ex et quam laborum",
      "postId": "1",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`[testing photo queries] GET_ONE_COMMENT with not existing comment 1`] = `
Object {
  "data": Object {
    "comment": Object {
      "error": "Error",
      "message": "Comment with the id 1000 not exists",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
